### PR TITLE
drivers: i2c: add Broadcom iProc I2C driver

### PR DESCRIPTION
**Pull request by ycsin**
Add device driver, bindings and build-only test for Broadcom iProc I2C.

### CODE REVIEW

**Review by teburd — DISMISSED** [review]
Nothing obviously wrong that I can see. Assuming testing has been done.

Would recommend testing with the recently added i2c ram test against sram/fram or similar to validate that the controller side functionality works as expected.

File: dts/bindings/i2c/brcm,iproc-i2c.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+properties:
+  reg:
+    required: true
+    description: |
+      Define the base and range of the I/O address space that contain the iProc
+      I2C controller registers
+
+  interrupts:
+    required: true
+    description: Define interrupt line number of the device
```
[the remark this answers is not in the document]
Comment (ycsin) [reply dts/bindings/i2c/brcm,iproc-i2c.yaml]: Removed them as not really required, thanks

**Comment by teburd** [comment]
Commit message has spi rather than i2c in its title, seems like an oops

**Comment by ycsin** [comment]
Updated commit message and updated the driver to use `I2C_DEVICE_DT_INST_DEFINE`

**Review by teburd — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**